client/src/components/Signup.js

```diff
@@ -184,13 +184,17 @@ export default class Signup extends React.Component {
             e.preventDefault();
             const endPoint = this.props.isAlumni ? '/alumni' : '/student';
             try {
-                const result = makeCall(payload, endPoint, 'post')
+                const result = await makeCall(payload, endPoint, 'post')
                 if (!result || result.error) {
-                    swal({
-                        title: "Error!",
-                        text: "There was an error completing your request, please try again.",
-                        icon: "error",
-                    });
+                    this.setState({
+                        submitting: false
+                    }, () => {
+                        swal({
+                            title: "Error!",
+                            text: "There was an error completing your request, please try again.",
+                            icon: "error",
+                        });
+                    })
                 } else {
                     this.setState({
                         submitting: false
```

Comment:
`this.setState({
                        submitting: false
                    }, () => {
                        swal({
                            title: "Error!",
                            text: "There was an error completing your request, please try again.",
                            icon: "error",
                        });
                    })
                } else {
                    this.setState({
                        submitting: false
`
Should the second setState set submitting to true if it's a successful submission?

Reply:
Yes! What `submitting` is doing is setting the loading state of the spinner on the submit button.